model/client4.go

```diff
@@ -2814,7 +2814,7 @@ func (c *Client4) GetAllChannelsWithCount(page int, perPage int, etag string) (C
 }
 
 // CreateChannel creates a channel based on the provided channel struct.
-func (c *Client4) CreateChannel(channel *Channel) (*Channel, *Response, error) {
+func (c *Client4) CreateChannel(channel *ChannelWithCategoryData) (*Channel, *Response, error) {
```

Comment:
Oops, sorry @anurag6713. We cannot make breaking changes to the model package. Please create a new method: `CreateChannelWithCategory`.